Makefile

```diff
@@ -13,10 +13,70 @@ GOPROXY ?= $(shell go env GOPROXY)
 # Specify additional `docker build` arguments. (Recipes: image)
 IMAGE_ARGS ?= -t hegel
 
+# Root output directory.
+OUT_DIR ?= $(shell pwd)/out
+
+# Project binary output directory.
+BIN_DIR ?= $(OUT_DIR)/bin
+
+# Linter installation directory.
+TOOLS_DIR ?= $(OUT_DIR)/tools
+
+# Platform variables
+PLATFORM_OS 	?= $(shell uname)
+PLATFORM_ARCH 	?= $(shell uname -m)
+
 .PHONY: help
 help:
 	@awk 'BEGIN {FS = ":.*##"; printf "Usage:\n  make \033[36m<target>\033[0m\n"} /^[%\/0-9A-Za-z_-]+:.*?##/ { printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 
+# -- Tooling
+
+GOLANGCI_LINT_VERSION 	?= v1.51.2
+GOLANGCI_LINT 			:= $(TOOLS_DIR)/golangci-lint
+$(GOLANGCI_LINT):
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
+		sh -s -- -b $(TOOLS_DIR) $(GOLANGCI_LINT_VERSION)
+
+# The command to run mockgen.
+MOCKGEN_VERSION 	?= v1.6.0
+MOCKGEN 			:= $(TOOLS_DIR)/mockgen
+$(MOCKGEN): PLATFORM_ARCH := $(if $(filter x86_64,$(PLATFORM_ARCH)),amd64,$(PLATFORM_ARCH))
+$(MOCKGEN): PLATFORM_OS := $(shell echo $(PLATFORM_OS) | tr '[:upper:]' '[:lower:]')
+$(MOCKGEN):
+	curl -sSfL https://github.com/golang/mock/releases/download/$(MOCKGEN_VERSION)/mock_$(MOCKGEN_VERSION:v%=%)_$(PLATFORM_OS)_$(PLATFORM_ARCH).tar.gz | \
+		tar -C $(TOOLS_DIR) --strip-components=1 -xzf -
+	@rm $(TOOLS_DIR)/README.md $(TOOLS_DIR)/LICENSE
+
+SETUP_ENVTEST := $(TOOLS_DIR)/setup-envtest
+$(SETUP_ENVTEST):
+	go build -tags=tools -o $@ sigs.k8s.io/controller-runtime/tools/setup-envtest
```

Comment:
I realized we can just do `GOBIN=<dir> go install ...` which would align better with all other tool installations and get rid of the `tools.go`.